src/app/public/modules/toggle/toggle.component.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  Component,
+  EventEmitter,
+  forwardRef,
+  Input,
+  Output,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef
+} from '@angular/core';
+
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator
+} from '@angular/forms';
+
+import {
+  SkyToggleChange
+} from './types';
+
+/**
+ * Provider Expression that allows sky-toggle to register as a ControlValueAccessor.
+ * This allows it to support [(ngModel)].
+ */
+// tslint:disable:no-forward-ref no-use-before-declare
+export const SKY_TOGGLE_CONTROL_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyToggleComponent),
+  multi: true
+};
+
+const SKY_TOGGLE_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyToggleComponent),
+  multi: true
+};
+
+// tslint:enable
+
+@Component({
+  selector: 'sky-toggle',
+  templateUrl: './toggle.component.html',
+  styleUrls: ['./toggle.component.scss'],
+  providers: [
+    SKY_TOGGLE_CONTROL_VALUE_ACCESSOR,
+    SKY_TOGGLE_VALIDATOR
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyToggleComponent implements ControlValueAccessor, Validator {
+
+  @Input()
+  public ariaLabel: string;
+
+  @Input()
+  public ariaLabelledBy: string;
+
+  @Input()
+  public disabled: boolean = false;
+
+  @Input()
+  public tabindex: number = 0;
+
+  @Output()
+  public toggleChanged: EventEmitter<SkyToggleChange> = new EventEmitter<SkyToggleChange>();
```

Comment:
Events that continuously emit should be in the present tense, `toggleChange`.
See: https://docs.blackbaud.com/skyux-team-docs/style#eventsoutputs